action.go

```diff
@@ -441,14 +441,23 @@ func (p pipeline) Run(c Conn) error {
 	if err := c.Encode(p); err != nil {
 		return err
 	}
-	for _, cmd := range p {
+
+	for i, cmd := range p {
 		if err := c.Decode(cmd); err != nil {
+			p.drain(c, len(p)-i-1)
 			return decodeErr(cmd, err)
 		}
 	}
 	return nil
 }
 
+func (p pipeline) drain(c Conn, n int) {
+	var rcv resp2.RawMessage
```

Comment:
Rather than a `resp2.RawMessage`, using a `resp2.Any{I: nil}` will discard the next message on the wire, rather than reading it into a buffer and incurring an allocation.

Reply:
Fixed! I'm amended the existing commit.